packages/labs/eleventy-plugin-lit/README.md

```diff
@@ -1,68 +1,396 @@
+bare module specifier and defines a component with `customElements.define`.
 
-```bash
+### Enable experimental VM modules
+
+> 🚧 Note: Removing this requirement is on the [roadmap](#roadmap). Follow
+> [#2494](https://github.com/lit/lit/issues/2484) for progress and discussion.
+> 🚧
+
+Eleventy _must_ be executed with the
+[`--experimental-vm-modules`](https://nodejs.org/api/vm.html#class-vmmodule)
+Node flag enabled. This flag is available in Node versions `12.16.0` and above.
+This flag allows the lit plugin to render components without modifying any
+global state, and enables compatibility with Eleventy's `--watch` mode.
+
+```sh
 NODE_OPTIONS=--experimental-vm-modules eleventy
 ```
 
-## Client-side hydration
+### Watch mode
 
-For hydrating interactive components on the client, you should include
-`lit/experimental-hydrate-support.js` along with the component definitions used
-(it may be included in the same bundle passed to the eleventy plugin if serving
-bundled sources to the client) _at the bottom of the page_. When server-rendering components, you should load definitions after components have been parsed.
+Use [`addWatchTarget`](https://www.11ty.dev/docs/watch-serve/) to tell Eleventy
+to watch for changes in your JavaScript directory:
 
-Most browsers (excluding Chrome 90+) will also need to load and invoke the `@webcomponents/template-shadowroot` ponyfill prior to registering component definitions:
+```js
+eleventyConfig.addWatchTarget('js/');
+```
 
-```html
-<script type="module">
-  // Hydrate template-shadowroots eagerly after rendering (for browsers without
-  // native declarative shadow roots)
-  import {
-    hasNativeDeclarativeShadowRoots,
-    hydrateShadowRoots,
-  } from './node_modules/@webcomponents/template-shadowroot/template-shadowroot.js';
-  if (!hasNativeDeclarativeShadowRoots) {
-    hydrateShadowRoots(document.body);
+## Usage
+
+Whenever you use a custom element in your Eleventy Markdown and HTML files,
+`@lit-labs/eleventy-plugin-lit` will automatically render its template and
+styles directly into your HTML.
+
+For example, given a markdown file `hello.md`:
+
+```md
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
+```
+
+And a component definition `js/demo-greeter.js`:
+
+<!-- prettier-ignore-start -->
+```js
+import {LitElement, html, css} from 'lit';
+
+class DemoGreeter extends LitElement {
+  static styles = css`
+    b { color: red; }
+  `;
+
+  static properties = {
+    name: {},
+  };
+
+  render() {
+    return html`Hello <b>${this.name}</b>!`;
   }
-  // ...
-  // Load and hydrate components lazily
-  import('./_js/components.bundle.js');
-</script>
+}
+customElements.define('demo-greeter', DemoGreeter);
+```
+<!-- prettier-ignore-end -->
+
+Then the Eleventy will produce `greeting/index.html`:
+
+<!-- prettier-ignore-start -->
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
```

Comment:
Maybe add a small note right below the sample that `<template shadowroot="open">` is an HTML standard for declarative shadow roots and either browser will automatically attach the shadow root (Chrome, Edge) or a polyfill will. With links.

Reply:
Done, but just a brief sentence that links to the Declarative shadow DOM section that we already have below.